src/vmm/src/lib.rs

```diff
@@ -908,17 +908,19 @@ impl MutEventSubscriber for Vmm {
             let mut exit_code = None;
             // Query each vcpu for their exit_code.
             for handle in &self.vcpus_handles {
-                match handle.response_receiver().try_recv() {
-                    Ok(VcpuResponse::Exited(status)) => {
-                        exit_code = Some(status);
-                        // Just use the first encountered exit-code.
+                // Drain all vcpu responses that are pending from this vcpu until we find an
+                // exit status.
+                for response in handle.response_receiver().try_iter() {
+                    if let VcpuResponse::Exited(status) = response {
+                        // It could be that some vcpus exited successfully while other's
+                        // errored out. Thus make sure that error exits from one vcpu always
+                        // takes precedence over "ok" exits exit_code.is_none()
+                        if exit_code.is_none() || exit_code == Some(FcExitCode::Ok) {
+                            exit_code = Some(status);
+                        }
+                        // We do not care about any other messages from this vcpu
                         break;
```

Comment:
I think this breaks on the 1st iteration of the for loop so we don't get to apply the logic of capturing the errors from other vcpus.

Reply:
Not really sure what you mean here :/

Reply:
ah nvm, there was a comment which clouded my vision :P and I thought we break from the for loop.